### ISSUE TITLE
Determine github associated with current git repo?

### ISSUE THREAD

**Issue by hadley** [r-pkgs/gh#40]
I'm not sure if this belongs here (esp. since it adds a dependency on git2r), but it wouldn't be an unreasonable home.

**Comment by jennybc** [comment]
Or githug which I can return to soon?

W/o using Git or git2r, you can get username from the token and make a reasonable guess at repo name.

**Comment by gaborcsardi** [comment]
Just read it from the .git/config file. (Or wherever GIT_DIR / GIT_CONFIG
point to.)

**Comment by hadley** [comment]
True, we wouldn't necessarily need a git2r dep, but there's quite a bit of code needed to detect if a remote is a github remote.

**Comment by gaborcsardi** [comment]
@hadley Is that because of GH enterprise? 

Can't you just take the URL from there, and 
* if a github.com URL, then it is GH
* otherwise try to query it with the GH API, and if it answers, it is GH?
No?

**Comment by hadley** [comment]
Hmmm I hadn't thought of that strategy - I think that helps a bit, but you still need to deal with parsing the `.config` file (maybe you have a package for that?) and ssh urls.

**Comment by gaborcsardi** [comment]
It is just an `ini` file, and yes, there is a package for it (`ini`), although I have never used it.

ssh urls cannot be too hard to parse, I bet there is a regex that covers practically everything.

I am more worried about git configuration, that is quite sophisticated with `GIT_DIR` and `GIT_CONFIG` and system, user and repo config files, etc.

But if it is "only" (sorry!) for `usethis`, it it is also not a tragedy if the detection fails for some weird configs...

**Comment by hadley** [comment]
It seems like it might be more generally useful to be able to access the "current" repo, which is was I was hoping you might be excited to add it to gh ;)

**Comment by gaborcsardi** [comment]
@hadley wait, I am not sure what you mean, then. The title says you want the GH remote(s) (if any) of the current repo. That is in `.git/config` (or ...).  No?

Considering that gh is for GH, yes, I can add this. (But I really don't want a `git2r` dependency, if possible.) Unless you want sg lse.

**Comment by hadley** [comment]
I think we're talking at cross-purposes — I have working code in devtools that we could use once we have a list of remotes. I think we can get a list of remotes by looking for `.git/config` and manually parsing with regexps, or add a dependency on git2r or ini.

If you'd be happy to have this in gh, I can do a PR.

**Comment by gaborcsardi** [comment]
> If you'd be happy to have this in gh, I can do a PR.

Sure, thanks much! `ini::read.ini` works fine, btw.
